main.tf

```diff
@@ -18,35 +18,38 @@
-  project                      = "${var.project_id != ""}"
-  boolean_policy               = "${var.policy_type == "boolean" }"
-  list_policy                  = "${var.policy_type == "list" && !local.invalid_config}"
-  enforce                      = "${(var.allow_list_length > 0 || var.deny_list_length > 0) ? "" : var.enforce}"
-  exclude_folders_list_length  = "${length(compact(var.exclude_folders))}"
-  exclude_projects_list_length = "${length(compact(var.exclude_projects))}"
-  invalid_config_case_1        = "${var.deny_list_length > 0 && var.allow_list_length > 0}"
-  invalid_config_case_2        = "${(var.allow_list_length + var.deny_list_length) > 0 && (var.enforce == "true" || local.enforce == "false")}"
-  invalid_config               = "${var.policy_type == "list" && (local.invalid_config_case_1 || local.invalid_config_case_2)}"
+  organization                 = var.organization_id != null
+  folder                       = var.folder_id != null
+  project                      = var.project_id != null
+  boolean_policy               = var.policy_type == "boolean"
+  list_policy                  = var.policy_type == "list" && ! local.invalid_config
+  enforce                      = var.allow_list_length > 0 || var.deny_list_length > 0 ? null : var.enforce
+  exclude_folders_list_length  = length(compact(var.exclude_folders))
+  exclude_projects_list_length = length(compact(var.exclude_projects))
+  invalid_config_case_1        = var.deny_list_length > 0 && var.allow_list_length > 0
+  invalid_config_case_2        = var.allow_list_length + var.deny_list_length > 0 && var.enforce != null
+  invalid_config               = var.policy_type == "list" && local.invalid_config_case_1 || local.invalid_config_case_2
 }
 
 /******************************************
   Checks a valid configuration for list constraint
  *****************************************/
 resource "null_resource" "config_check" {
   /*
-     * This resource shows the user a message intentionally
-     *
-     * If user sets two (or more) of following variables when policy type is "list":
-     * - allow
-     * - deny
-     * - enforce ("true" or "false")
-     * the configuration is invalid and the message below is shown
-     *
-     * ( This approach is taken from https://github.com/hashicorp/terraform/issues/15469 )
-     */
-  count = "${local.invalid_config ? 1 : 0}"
+    This resource shows the user a message intentionally
+    If user sets two (or more) of following variables when policy type is "list":
+    - allow
+    - deny
+    - enforce ("true" or "false")
+    the configuration is invalid and the message below is shown
+  */
+  count = local.invalid_config ? 1 : 0
 
-  "For list constraints only one of enforce, allow, and deny may be included." = true
+  provisioner "local-exec" {
+    command = "echo 'For list constraints only one of enforce, allow, and deny may be included.'; false"
+  }
+}
+
+terraform {
```

Comment:
This block should be moved to `versions.tf` per the convention set by `terraform 0.12upgrade`.

Reply:
Fixed last one. @aaron-lane, PR is ready for review